api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -60,4 +60,9 @@
    * @return this for method chaining
    */
   ThisT scanManifestsWith(ExecutorService executorService);
+
+  /**
+   * Perform operations a particular branch
```

Comment:
This needs a `@param` to document branch. I think we also need to specify behavior:
* What happens if the branch does not exist? I think we would create one from the `main` branch.
* How does this change validation? What if the validation's from snapshot is not an ancestor of the branch?

Reply:
@rdblue made changes to to parameter to take the type SnapshotRef so that end customer will be passing a valid snapshot ref.

Reply:
@rdblue  i added a check to check if branch exists. If it does not exist should we create new branch? I was thinking to throw an illegal arg exception.

_How does this change validation? What if the validation's from snapshot is not an ancestor of the branch? - > I did not understand this scenario._

Reply:
Each child of SnapshotProducer can implement their own validation logic https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SnapshotProducer.java#L156

This gets called when trying to apply the operation as part of transaction conflict detection

Reply:
@amogh-jahagirdar All validations must start from tip of branch, aka the current snapshot in the branch. So all the operations which override the ` protected void validate(TableMetadata currentMetadata) { }` aka - BaseOverwriteFiles, BaseReplacePartitions, BaseRewriteFiles, BaseRowDelta need to reference the current snapshot from the branch tip. 

Is any other validation needed here ?